File: android/src/main/kotlin/com/github/wratheus/flutter_simple_ble/BleChannelContract.kt
// SPDX-FileCopyrightText: 2026 Aleksandr <https://github.com/Wratheus>
// SPDX-License-Identifier: BSD-3-Clause

package com.github.wratheus.flutter_simple_ble

internal object BleChannelContract {
    const val CHANNEL_NAME = "flutter_simple_ble"

    object Method {
        const val IS_SUPPORTED = "isSupported"
        const val GET_ADAPTER_STATE = "getAdapterState"
        const val START_SCAN = "startScan"
        const val STOP_SCAN = "stopScan"
        const val CONNECT = "connect"
        const val DISCONNECT = "disconnect"
        const val DISCOVER_SERVICES = "discoverServices"
        const val REQUEST_MTU = "requestMtu"
        const val WRITE_CHARACTERISTIC = "writeCharacteristic"
    }

    object Event {
        const val ADAPTER_STATE_CHANGED = "OnAdapterStateChanged"
        const val SCAN_RESPONSE = "OnScanResponse"
        const val CONNECTION_STATE_CHANGED = "OnConnectionStateChanged"
        const val DISCOVERED_SERVICES = "OnDiscoveredServices"
        const val MTU_CHANGED = "OnMtuChanged"
        const val CHARACTERISTIC_WRITTEN = "OnCharacteristicWritten"
    }

    object Key {
        const val ADAPTER_STATE = "adapter_state"
        const val ADVERTISEMENTS = "advertisements"
        const val REMOTE_ID = "remote_id"
        const val PLATFORM_NAME = "platform_name"
        const val ADV_NAME = "adv_name"
        const val RSSI = "rssi"
        const val CONNECTABLE = "connectable"
        const val TX_POWER_LEVEL = "tx_power_level"
        const val APPEARANCE = "appearance"
        const val MANUFACTURER_DATA = "manufacturer_data"
        const val SERVICE_DATA = "service_data"
        const val SERVICE_UUIDS = "service_uuids"
        const val CONNECTION_STATE = "connection_state"
        const val DISCONNECT_REASON_CODE = "disconnect_reason_code"
        const val DISCONNECT_REASON_STRING = "disconnect_reason_string"
        const val SERVICES = "services"
        const val PRIMARY_SERVICE_UUID = "primary_service_uuid"
        const val SERVICE_UUID = "service_uuid"
        const val CHARACTERISTICS = "characteristics"
        const val CHARACTERISTIC_UUID = "characteristic_uuid"
        const val INSTANCE_ID = "instance_id"
        const val PROPERTIES = "properties"
        const val BROADCAST = "broadcast"
        const val READ = "read"
        const val WRITE_WITHOUT_RESPONSE = "write_without_response"
        const val WRITE = "write"
        const val NOTIFY = "notify"
        const val INDICATE = "indicate"
        const val AUTHENTICATED_SIGNED_WRITES = "authenticated_signed_writes"
        const val EXTENDED_PROPERTIES = "extended_properties"
        const val MTU = "mtu"
        const val VALUE = "value"
        const val WRITE_TYPE = "write_type"
        const val SUCCESS = "success"
        const val ERROR_CODE = "error_code"
        const val ERROR_STRING = "error_string"
    }
}

File: ios/flutter_simple_ble/Sources/flutter_simple_ble/BleChannelContract.swift
// SPDX-FileCopyrightText: 2026 Aleksandr <https://github.com/Wratheus>
// SPDX-License-Identifier: BSD-3-Clause

enum BleChannelContract {
    static let channelName = "flutter_simple_ble"

    enum Method {
        static let isSupported = "isSupported"
        static let getAdapterState = "getAdapterState"
        static let startScan = "startScan"
        static let stopScan = "stopScan"
        static let connect = "connect"
        static let disconnect = "disconnect"
        static let discoverServices = "discoverServices"
        static let requestMtu = "requestMtu"
        static let writeCharacteristic = "writeCharacteristic"
    }

    enum Event {
        static let adapterStateChanged = "OnAdapterStateChanged"
        static let scanResponse = "OnScanResponse"
        static let connectionStateChanged = "OnConnectionStateChanged"
        static let discoveredServices = "OnDiscoveredServices"
        static let mtuChanged = "OnMtuChanged"
        static let characteristicWritten = "OnCharacteristicWritten"
    }

    enum Key {
        static let adapterState = "adapter_state"
        static let advertisements = "advertisements"
        static let remoteId = "remote_id"
        static let platformName = "platform_name"
        static let advName = "adv_name"
        static let rssi = "rssi"
        static let connectable = "connectable"
        static let txPowerLevel = "tx_power_level"
        static let manufacturerData = "manufacturer_data"
        static let serviceData = "service_data"
        static let serviceUuids = "service_uuids"
        static let connectionState = "connection_state"
        static let disconnectReasonCode = "disconnect_reason_code"
        static let disconnectReasonString = "disconnect_reason_string"
        static let services = "services"
        static let primaryServiceUuid = "primary_service_uuid"
        static let serviceUuid = "service_uuid"
        static let characteristics = "characteristics"
        static let characteristicUuid = "characteristic_uuid"
        static let instanceId = "instance_id"
        static let properties = "properties"
        static let broadcast = "broadcast"
        static let read = "read"
        static let writeWithoutResponse = "write_without_response"
        static let write = "write"
        static let notify = "notify"
        static let indicate = "indicate"
        static let authenticatedSignedWrites = "authenticated_signed_writes"
        static let extendedProperties = "extended_properties"
        static let mtu = "mtu"
        static let value = "value"
        static let writeType = "write_type"
        static let success = "success"
        static let errorCode = "error_code"
        static let errorString = "error_string"
    }
}
